internal/mssql/factory.go

```diff
@@ -0,0 +1,9 @@
+package mssql
+
+func New(unitTesting bool) MssqlInterface {
```

Comment:
>fun [](http://example.com/codeflow?start=0&length=3)

it's a little weird to me for the production code and the test code to live in the same binary.

Could the mock implementation only live in a _test.go file and the factory accept an optional delegate used to create the instance? If no delegate is passed, it uses the default production type. 
The unit tests would pass a delegate that creates the mock.
Alternatively we could use a build tag to control inclusion of the mock type and make sure the official product build sets the tag to exclude it.

Reply:
I've made a note to talk about this at our next sync.  I'm doing this to speed up unit testing (of other units), I think the mock impl needs to be available in prod code here, but let's chat